File: library/src/main/java/obby0511/sentiment/analysis/LanguageService.java
package obby0511.sentiment.analysis;

import lombok.Builder;
import lombok.Getter;
import lombok.Value;

import java.nio.charset.Charset;

public interface LanguageService {

    LanguageResponse analyzeSentiment(LanguageRequest request);

    @Value
    @Builder
    class LanguageRequest {
        private String text;
        private Charset encodingType;
    }

    @Getter
    class LanguageResponse {
        private final float score;
        private final double magnitude;

        public LanguageResponse(double magnitude) {
            this.magnitude = magnitude;
            this.score = magnitude < 0 ? -1
                    : magnitude > 0 ? 1
                    : 0;
        }
    }
}

File: src/main/kotlin/obby0511/sentiment/analysis/domain/service/LanguageService.kt
package obby0511.sentiment.analysis.domain.service

interface LanguageService {

  fun analyzeEntitySentiment(request: AnalyzeEntitySentimentRequest): AnalyzeEntitySentimentResponse

}

data class AnalyzeEntitySentimentRequest(val document: String, val encodingType: Charsets) {
}

data class AnalyzeEntitySentimentResponse(val magnitude: Float, val score: Float) {
}
